src/ActivitiesSDKAdapter.js

```diff
@@ -168,4 +136,64 @@ export default class ActivitiesSDKAdapter extends ActivitiesAdapter {
 
     return activity$;
   }
+
+  /**
+   * A function that checks whether or not an Activity object contains a card attachment.
+   *
+   * @param {Activity} activity  Activity object
+   * @returns {boolean} True if received Activity object contains a card attachment
+   */
+  // eslint-disable-next-line class-methods-use-this
+  hasAdaptiveCard(activity) {
+    return !!activity.attachments && activity.attachments[0] && activity.attachments[0].contentType === 'application/vnd.microsoft.card.adaptive';
```

Comment:
let's make sure that this only returns true or false. You did it for attachments (if missing, false is returned) but if the first attachment is missing, undefined is returned.

```js
return !!(activity.attchments && activity.attachments[0] && ...);